### ISSUE TITLE
SHOPT_DYNAMIC=1 build leads to path.sh regression

### ISSUE THREAD

**Issue by phidebian** [ksh93/ksh#673]
Building with 
````
./bin/package make \
    CCFLAGS='-m64 -O0 -g3 -DSHOPT_DYNAMIC=1' \
    LDFLAGS='-m64 -g3'
````

And testing with `./bin/package test` lead to

````
Total errors: 3
CPU time       user:      system:
main:      0m00.178s    0m00.149s
tests:     1m24.329s    0m36.588s
````

Running `./arch/linux.i386-64/bin/ksh`  exhibit the following
behavior.

````
TC$ command -vx cat                                                           
/usr/bin/cat

TC$ PATH=/opt/ast/bin:$PATH

TC$ command -vx cat        
/opt/ast/bin/cat

TC$ command -vx cat
/usr/bin/cat
````

Adding `/opt/ast/bin` in the `PATH` makes the `command -vx cat`
wrongly report `/opt/ast/bin/cat` then subsequent display the correct `/usr/bin/cat`

**Comment by McDutchie** [comment]
Tested and confirmed (finally – sorry about that).

**Comment by McDutchie** [comment]
Fixed in e63febac3a45d4bd8f7e06ec2b4464af0d23b377 and fix improved in 785c6b98943eb4957f6c5792eeed391f75b9b702.
